apps/documents/views.py

```diff
@@ -349,3 +349,49 @@ def form_valid(self, form):
             assistant_id=assistant.id,
         )
         return HttpResponseRedirect(self.get_success_url())
+
+
+class FileChunkEmbeddingListView(LoginAndTeamRequiredMixin, ListView, PermissionRequiredMixin):
+    """View to display file chunks for a specific file in a collection with pagination"""
+
+    model = FileChunkEmbedding
+    template_name = "documents/file_chunks.html"
+    context_object_name = "chunks"
+    paginate_by = 10
+    permission_required = ("documents.view_collection", "files.view_file")
+
+    def get_queryset(self):
+        collection_id = self.kwargs["collection_id"]
+        file_id = self.kwargs["file_id"]
+
+        # Get chunks for this file in this collection, ordered by chunk number
+        return FileChunkEmbedding.objects.filter(
+            collection_id=collection_id, file_id=file_id, is_archived=False
```

Comment:
This should also filter by team just to be safe

Reply:
https://github.com/dimagi/open-chat-studio/pull/1772/commits/7593e11431639c009d5f770e7f54d4caed087c68